### PR TITLE
E2E code cleanup

### PR DESCRIPTION
**Pull request by alfonsomthd**
- Move cypress deps. to devDependencies.
- ts-node script: remove unneeded 'ignore' flag. 
- Remove the data-test-id selectors.
- Add 'ByTestSelector' method.
- Add the cypress folder to lint commands.
- New yarn command "test-cypress-with-op-install" that also installs operator.
- E2E CI scripts: install operator before running cypress.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
     "cypress-generate": "marge -o ./cypress-gen/ -f cypress-report -t 'OpenShift Console Cypress Test Results' -p 'OpenShift Cypress Test Results' --showPassed false --assetsDir ./cypress-gen/cypress/assets ./cypress-gen/cypress.json",
     "cypress-postreport": "yarn cypress-merge && yarn cypress-generate",
     "test-cypress": "node_modules/.bin/cypress open --config-file ./cypress/cypress.json --env openshift=true",
+    "test-cypress-ci": "./e2e-ops.sh && node_modules/.bin/cypress open --config-file ./cypress/cypress.json --env openshift=true",
```
Comment (SanjalKatiyar) [inline package.json:13]: just want to confirm is naming is correct... `test-cypress-ci` make me feel like this commands gets executed in CI, but this is only needed in local...

Comment (alfonsomthd) [reply package.json]: > Maybe `test-cypress-local`?

`local` can be also misleading as plain `test-cypress` is meant to run in local...

Following descriptive naming I think that `test-cypress-with-op-install` reveals its purpose.

**Pull request merged**